src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -351,6 +351,22 @@ inline std::istream& operator>>(std::istream& is, PerformanceMode& performance_m
  */
 static constexpr Property<PerformanceMode> performance_mode{"PERFORMANCE_HINT"};
 
+/**
+ * @brief This property define if using hyper threading during inference.
+ * @ingroup ov_runtime_cpp_prop_api
+ *
+ * Developer can use this property to use or not use CPU pinning during inference. If user does not explicitly set
+ * value for this property, OpenVINO may choose any desired value based on internal logic.
+ *
+ * The following code is example to use this property.
+ *
+ * @code
+ * ie.set_property(ov::hint::use_cpu_pinning(true));
+ * ie.set_property(ov::hint::use_cpu_pinning(false));
+ * @endcode
+ */
+static constexpr Property<bool> use_cpu_pinning{"USE_CPU_PINNING"};
```

Comment:
is it supposed that we need to deprecate AFFINITY?

Reply:
Yes, we need to document AFFINITY deprecation in 2023.0 and deprecate AFFINITY in 2024.0.

Reply:
Why in 2024?
We can deprecate even now.

Is it expected to fully replace affinity by use_cpu_pinning ? Affinity has more values like NUMA, CORES than just true / false in use_cpu_pinning

Reply:
Could we deprecate property without information in advance? 

From technique perspective, yes, we will replace affinity by use_cpu_pinning now.